pkg/validation/sessions.go

```diff
@@ -38,6 +43,96 @@ func validateSessionCookieMinimal(o *options.Options) []string {
 	return msgs
 }
 
+func validateDynamoDBSessionStore(o *options.Options) []string {
+	if o.Session.Type != options.DynamoDBSessionType {
+		return []string{}
+	}
+
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	if err != nil {
+		return []string{fmt.Sprintf("unable to create dynamoDB client: %v", err)}
+	}
+
+	nonce, err := encryption.Nonce()
+	if err != nil {
+		return []string{fmt.Sprintf("unable to generate a dynamoDB initialization test key: %v", err)}
+	}
+
+	key := fmt.Sprintf("%s-healthcheck-%s", o.Cookie.Name, nonce)
+	return sendDynamoDBConnectionTest(dynamo.New(sess), o.Session.DynamoDB.TableName, key, nonce)
+}
+
+func sendDynamoDBConnectionTest(service *dynamo.DynamoDB, tableName string, key string, nonce string) []string {
+	ctx := context.Background()
+	msgs := []string{}
+
+	av, err := dynamodbattribute.MarshalMap(dynamodb.DynamoSessionItem{
+		SessionKey: key,
+		Value:      []byte(nonce),
+		Expiry:     "",
+	})
+	if err != nil {
+		msgs = append(msgs, fmt.Sprintf("error saving dynamodb initialisation key: %v", err))
+		return msgs
+	}
+
+	_, err = service.PutItemWithContext(ctx, &dynamo.PutItemInput{
+		Item:      av,
+		TableName: aws.String(tableName),
+	})
+	if err != nil {
+		msgs = append(msgs, fmt.Sprintf("unable to save dynamodb initialisation key: %v", err))
+		return msgs
+	}
+
+	result, err := service.GetItemWithContext(ctx, &dynamo.GetItemInput{
+		TableName: aws.String(tableName),
+		Key: map[string]*dynamo.AttributeValue{
+			dynamodb.SessionKeyName: {
+				S: aws.String(key),
+			},
+		},
+		ConsistentRead: aws.Bool(true),
+	})
+	if err != nil {
+		msgs = append(msgs, fmt.Sprintf("error reading dynamodb initialisation key: %v", err))
+		return msgs
+	}
+
+	item := dynamodb.DynamoSessionItem{}
+	if err := dynamodbattribute.UnmarshalMap(result.Item, &item); err != nil {
+		msgs = append(msgs, fmt.Sprintf("error reading dynamodb initialisation key: %v", err))
+		return msgs
```

Comment:
I don't think we want to return at this point yet.  We still want to attempt to delete at the end to keep things clean and not pollute the table.

Reply:
(Same note for the other `Load` flavored errors above. Potentially the initial `Save` as well...? We can return immediately if that fails if you want)